### PR TITLE
[eas-cli] add eas update:configure

### PR DESCRIPTION
**Pull request by jkhales**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

We want a way to auto configure the EAS Update URL.

# How

Added command `eas update:configure`

# Test Plan

tested in demo repo.

### CODE REVIEW

**Comment by jkhales** [comment]
/changelog-entry new-feature Add command: 'eas update:configure'

**Review by dsokal — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]
We maintain part of the updates configuration inside `eas build:configure` command, as far as I remember it's only generating Expo.plist/modifies AndroidManifest.xml for bare project.

If we are adding `updates:configure` then maybe that logic should be moved here? Technically code in build command also configures classic updates, but if we plan to maintain them indefinitely then keeping logic responsible for classic updates in  `update:configure` makes more sense.

File: packages/eas-cli/src/commands/update/configure.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    const hasAndroidNativeProject =
+      (await resolveWorkflowAsync(projectDir, Platform.ANDROID)) === Workflow.GENERIC;
+    const hasIosNativeProject =
+      (await resolveWorkflowAsync(projectDir, Platform.IOS)) === Workflow.GENERIC;
+
+    Log.addNewLineIfNone();
+    if (hasAndroidNativeProject || hasIosNativeProject) {
+      Log.log(
+        `🧐 It seems you are on the bare workflow! Please also update your native files. You can do this by either running 'eas build:configure' or manually editing the Expo.plist/AndroidManifest.xml. For details check: ${learnMore(
```
[the remark this answers is not in the document]
Comment (jkhales) [reply packages/eas-cli/src/commands/update/configure.ts:78]: good point, edited your suggestion to include this.

**Comment by jkhales** [comment]
> 

Yea, great question. I went back and forth with this. 

The issue is that this then triggers a question about which platforms you want to configure the native files for, which is misleading/confusing for the managed case, where only the `app.json` is modified. After talking with @jonsamp about the UX, decided to follow the path of least change for the first pass and maybe add more later.

**Pull request merged**